lib/Models/Leaflet.js

```diff
@@ -201,6 +201,36 @@ Leaflet.prototype.computePositionOnScreen = function(position, result) {
     return result;
 };
 
+
+
+/**
+ * Add and remove attribution text.
+ * @param  {textAttribution} the textAttribution object of each Catalog item.
+ */
+
+Leaflet.prototype.addAttributionText = function(textAttribution){
+    if(textAttribution){
+        var a = textAttribution.link?("<a href='" + textAttribution.link +"'>" + textAttribution.text + "</a>") : textAttribution.text;
+        addAttributionControl(this.map);
+        this.map.addControl(this.map.attributionControl);
+        this.map.attributionControl.setPrefix(a);
+    }else{
+        return false;
+    }
+};
+
+Leaflet.prototype.removeAttributionText = function(textAttribution){
+    if(textAttribution){
+        this.map.attributionControl.removeFrom(this.map);
```

Comment:
The add/remove logic isn't working quite right in Leaflet.  I think you want to update the existing `attributionControl` instead of creating a new one.  If you switch to 2D and then enable a layer with attribution, the text gets added to a separate control above the first one.  If you enable the layer while in 3D mode, and then switch to 2D, the attribution is instead added to the main attribution control (which is what we want, I'd say), but then when you disable the layer all attributions disappear, not just the ones for that layer.